google/cloud/documentai_toolbox/wrappers/page.py

```diff
@@ -186,56 +206,101 @@ class FormField:
+        _page (Page):
+            Required. The Page object.
     """
 
-    result_text = ""
+    documentai_object: documentai.Document.Page.Paragraph = dataclasses.field(
+        repr=False
+    )
+    _page: "Page" = dataclasses.field(repr=False)
 
-    for text_segment in layout.text_anchor.text_segments:
-        result_text += text[int(text_segment.start_index) : int(text_segment.end_index)]
+    lines: List[Line] = dataclasses.field(init=False, repr=False)
+    _text: Optional[str] = dataclasses.field(init=False, default=None)
+    _hocr_bounding_box: Optional[str] = dataclasses.field(init=False, default=None)
 
-    return result_text
+    def __post_init__(self):
+        self.lines = _get_children_of_element(self.documentai_object, self._page.lines)
 
+    @property
+    def text(self):
+        if self._text is None:
+            self._text = _text_from_layout(
+                layout=self.documentai_object.layout, text=self._page.text
+            )
+        return self._text
+
+    @property
+    def hocr_bounding_box(self):
+        if not self._hocr_bounding_box:
+            self._hocr_bounding_box = _get_hocr_bounding_box(
+                element_with_layout=self.documentai_object,
+                dimension=self._page.documentai_object.dimension,
+            )
+        return self._hocr_bounding_box
 
-def _trim_text(text: str) -> str:
-    r"""Remove extra space characters from text (blank, newline, tab, etc.)
 
-    Args:
-        text (str):
-            Required. UTF-8 encoded text in reading order
-            from the document.
-    Returns:
-        str:
-            Text without trailing spaces/newlines
+@dataclasses.dataclass
+class Block:
+    """Represents a wrapped documentai.Document.Page.Block.
+
+    Attributes:
+        documentai_object (google.cloud.documentai.Document.Page.Block):
+            Required. The original google.cloud.documentai.Document.Page.Block object.
+        _page (Page):
+            Required. The Page object.
     """
-    return text.strip().replace("\n", " ")
+
+    documentai_object: documentai.Document.Page.Block = dataclasses.field(repr=False)
+    _page: "Page" = dataclasses.field(repr=False)
+
+    paragraphs: List[Paragraph] = dataclasses.field(init=False, repr=False)
+    _text: Optional[str] = dataclasses.field(init=False, default=None)
+    _hocr_bounding_box: Optional[str] = dataclasses.field(init=False, default=None)
+
+    def __post_init__(self):
+        self.paragraphs = _get_children_of_element(
+            self.documentai_object, self._page.paragraphs
+        )
+
+    @property
+    def text(self):
+        if self._text is None:
+            self._text = _text_from_layout(
+                layout=self.documentai_object.layout, text=self._page.text
+            )
+        return self._text
+
+    @property
+    def hocr_bounding_box(self):
+        if not self._hocr_bounding_box:
```

Comment:
Let's be consistent and go with `if self._hocr_bounding_box is None` throughout.